src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/DirectoryRequest.cs

```diff
@@ -2,229 +2,121 @@
-
-            objClassAttr.Name = "objectClass";
-            objClassAttr.Add(objectClass);
-            _attributeList.Add(objClassAttr);
-        }
-
-        // Properties
-        public string DistinguishedName
-        {
-            get
             {
-                return _dn;
+                throw new ArgumentNullException(nameof(objectClass));
             }
+            
+            DistinguishedName = distinguishedName;
 
-            set
-            {
-                _dn = value;
-            }
-        }
-
-        public DirectoryAttributeCollection Attributes
-        {
-            get
+            var objClassAttr = new DirectoryAttribute()
             {
-                return _attributeList;
-            }
+                Name = "objectClass"
+            };
+            objClassAttr.Add(objectClass);
+            Attributes.Add(objClassAttr);
         }
 
-        //
-        // Private/protected
-        //
+        public string DistinguishedName { get; set; }
 
-        private string _dn;
-        private DirectoryAttributeCollection _attributeList;
+        public DirectoryAttributeCollection Attributes { get; } = new DirectoryAttributeCollection();
     }
 
     public class ModifyRequest : DirectoryRequest
     {
-        //
-        // Public
-        //
-        public ModifyRequest()
-        {
-            _attributeModificationList = new DirectoryAttributeModificationCollection();
-        }
+        public ModifyRequest() { }
 
         public ModifyRequest(string distinguishedName, params DirectoryAttributeModification[] modifications) : this()
         {
-            // Store off the distinguished name
-            _dn = distinguishedName;
-
-            // Store off the initial list of modifications
-            _attributeModificationList.AddRange(modifications);
+            DistinguishedName = distinguishedName;
+            Modifications.AddRange(modifications);
         }
 
         public ModifyRequest(string distinguishedName, DirectoryAttributeOperation operation, string attributeName, params object[] values) : this()
         {
-            // Store off the distinguished name
-            _dn = distinguishedName;
-
-            // validate the attributeName
             if (attributeName == null)
-                throw new ArgumentNullException("attributeName");
-
-            DirectoryAttributeModification mod = new DirectoryAttributeModification();
-            mod.Operation = operation;
-            mod.Name = attributeName;
-            if (values != null)
             {
-                for (int i = 0; i < values.Length; i++)
-                    mod.Add(values[i]);
+                throw new ArgumentNullException(nameof(attributeName));
             }
 
-            _attributeModificationList.Add(mod);
-        }
-
-        // Properties
-        public string DistinguishedName
-        {
-            get
+            DistinguishedName = distinguishedName;
```

Comment:
That was deliberate - it seems to be the convention that argument validation comes before assignment or the business logic